### PR TITLE
Add Ci and automated code formatter as well as linting tool

### LINKED ISSUE #49: Add CI for automated testing

### PR DESCRIPTION
**Pull request by t-ober**
Resolves #49

With flake8 you can test your code formatting from the command line with `flake8`.
With black typing `black .` in the command line should resolve all code formatting issues (this works like spotless).

### CODE REVIEW

**Review by sebastian-peter — CHANGES_REQUESTED** [review]
If I may express a wish, it'd be to also inlcude [isort](https://github.com/PyCQA/isort) :)
It can be [configured](https://pycqa.github.io/isort/docs/configuration/black_compatibility.html) to nicely play together with black.

File: .flake8 [hunk 1]
```diff
@@ -0,0 +1,4 @@
+[flake8]
```
[the remark this answers is not in the document]
Comment (t-ober) [reply .flake8:1]: I did adjust the configuration as suggested in the [documentation](https://black.readthedocs.io/en/stable/guides/using_black_with_other_tools.html) to make them work together

File: docs/sphinx/conf.py [hunk 2]
```diff
@@ -17,14 +17,17 @@
...
-            'Research group Distribution grid planning and operation'
-author = 'Johannes Hiry, Chris Kittl, Jawana Gabrielski, Nils Offermann, Thomas Oberließen'
+project = "Markovs Household"
+copyright = (
+    "© 2020. TU Dortmund University, Institute of Energy Systems, Energy Efficiency and Energy Economics, "
+    "Research group Distribution grid planning and operation"
+)
+author = (
+    "Johannes Hiry, Chris Kittl, Jawana Gabrielski, Nils Offermann, Thomas Oberließen"
+)
```
[the remark this answers is not in the document]
Comment (t-ober) [reply docs/sphinx/conf.py:27]: True

[the remark this answers is not in the document]
Comment (t-ober) [reply docs/sphinx/conf.py:27]: Would have to look a little further. Have opened an issue https://github.com/ie3-institute/MarkovsHousehold/issues/58

File: pyproject.toml [hunk 3]
```diff
@@ -13,6 +13,8 @@ pytest = "^7.1.2"
 PyYAML = "^6.0"
 black = "^22.3.0"
 flake8 = "^4.0.1"
+isort = "^5.10.1"
+flake8-isort = "^4.1.1"
```
Comment (sebastian-peter) [inline pyproject.toml:17]: Wouldn't these better fit inside `dev-dependencies` below? Which also applies to all other dependencies here besides `python` and `pyyaml`, I think

File: .flake8 (hunk 1 above)
[the remark this answers is not in the document]
Comment (sebastian-peter) [reply .flake8:1]: Hmm, well black and flake8 do not _conflict_ now, in the sense that one does not reverse changes of the other or vice versa.
But, they do not use the same code style for checking/correcting. Black is applying stricter rules than flake8 is checking for. So to make them consistent, I thought of using the plugin above... Kind of like `spotlessCheck` complains about style violations _excatly_ when `spotlessApply` would fix these violations.

**Review by sebastian-peter — CHANGES_REQUESTED** [review]
Some minor remarks

File: .github/workflows/simple-ci.yml [hunk 4]
```diff
@@ -32,10 +32,10 @@ jobs:
           poetry-version: ${{ matrix.poetry-version }}
       - name: Install poetry dependencies
         run: poetry install
-      - name: Lint with flake8
+      - name: Lint with flake8 / black
         run: |
          # stop the build if there are Python syntax errors or undefined names
-         poetry run flake8 . --count --show-source --statistics
+         poetry run black --check .
```
Comment (sebastian-peter) [inline .github/workflows/simple-ci.yml:38]: Add flake8 again since black does not cover all its tests?

**Review by sebastian-peter — APPROVED** [review]
Thanks!

**Pull request merged**